src/pluginrc/saltplugins.rc

```diff
@@ -1,17 +1,16 @@
 #!/bin/bash
 
 function check_packages() {
-    RPMS=$1
-    for pkg in $RPMS; do
-	if ! rpm -q $pkg &>/dev/null; then
-	    echo "ERROR: Salt package \"$pkg\" is not installed"
-	    echo
-	    exit 111
+    for pkg in "$@"; do
+	if ! validate_rpm_pkg $pkg; then
+	    exit 111 
 	fi
     done
 }
 
 validate_rpm_pkg() {
     OUT=$(rpm -V $1 2>&1)
-    [[ $? == 0 ]] && echo "Status: Passed" || echo "Status: $OUT"
+    R=$?
+    [[ ${R} == 0 ]] && echo "Status: Passed" || echo "Status: $OUT"
```

Comment:
Is there a reason for splitting this up?

Reply:
Yes, R is used one more time after these lines:
return ${R}

Reply:
one more point missing: $? changing the value on calling test inside the condition clause and the only way is to store the value before it

Reply:
Ups, totally missed that. Thx!